src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -1763,4 +1769,88 @@ namespace winrt::TerminalApp::implementation
 
         return Title();
     }
+
+    // Method Description:
+    // - Toggle broadcasting input to all the panes in this tab.
+    void TerminalTab::ToggleBroadcastInput()
+    {
+        _tabStatus.IsInputBroadcastActive(!_tabStatus.IsInputBroadcastActive());
+
+        const bool isBroadcasting = _tabStatus.IsInputBroadcastActive();
+        _rootPane->EnableBroadcast(isBroadcasting);
+
+        auto weakThis{ get_weak() };
+
+        // When we change the state of broadcasting, add or remove event
+        // handlers appropriately, so that controls won't be propagating events
+        // needlessly if no one is listening.
+
+        _rootPane->WalkTree([&](const auto& p) {
+            const auto paneId = p->Id();
+            if (!paneId.has_value())
+            {
+                return;
+            }
+            if (const auto& control{ p->GetTerminalControl() })
+            {
+                auto it = _controlEvents.find(*paneId);
+                if (it != _controlEvents.end())
+                {
+                    auto& events = it->second;
+
+                    if (isBroadcasting)
+                    {
+                        // ADD EVENT HANDLERS HERE
+                        events.keySentToken = control.KeySent([weakThis](auto&& sender, auto&& e) {

```

Comment:
in theory if we use auto revokers, we could simplify the code to this:

```
events.keySentToken.revoke();
xxx.revoke();

if (newIsBroadcasting) {
    keySentToken = foo;
    xxx = bar;
}
```

and simplify the else case plus stomp the event handlers properly every time